analysis_for_run.py

```diff
@@ -0,0 +1,236 @@
+    output, error = process.communicate()
+    tmp=output.decode("utf-8")
+    tmp=tmp.split('\n')
+    tmp.pop(0)
+    tmp.pop(-1)
+    tmp.pop(-1)
+    tmp=[x.replace('\t', ' ') for x in tmp]
+    return tmp
+
+# Function to loop through files needed to be downloaded for each analysis.
+def ica_download(target_dir,files_list):
+    for i in files_list:
+        bashCommand = "icav2 projectdata download %s%s /Volumes/IDGenomics_NAS/WGS_Serotyping/%s/" % (target_dir,i,args.run_name)
+        process = subprocess.Popen(bashCommand.split(), stdout=subprocess.PIPE)
+        process.communicate()
+
+# Slack_sdk values needed for sending messages to Slack. Uses an API already set up on the Slack website for the UPHL Workspace to post messages on the notifications channel.
+# Function makes sending Slack messages as easy as using the print funcition.
+client = WebClient(token=config.token)
+channel_id = config.channel_id
+def slack_message(string):
+    try:
+
+        result = client.chat_postMessage(
+            channel=channel_id,
+            text=string
+        )
+
+    except:
+        print("Slack Error")
+
+slack_message('Monitoring %s on BSSH' % args.run_name)
+print('Monitoring %s on BSSH' % args.run_name)
+# This While loops Uses the BaseSpace CLI tool to monitor the progress of the run
+t=0
+while t==0:
+    for i in configurations:
+        bashCommand='bs list --config=%s runs' % i
+        tmp=bs_out(bashCommand)
+        tmp=pd.DataFrame(index=[row.split()[2] for row in tmp[1:]], columns=tmp[0].split()[1:], data=[row.split()[1:4] for row in tmp[1:]])
+        try:
+            if tmp.at[args.run_name,'Status']=='Complete':
+                user=i
+                idd=tmp.at[args.run_name,'Id']
+                slack_message('%s is "Complete" on BSSH' % args.run_name)
+                print('%s is "Complete" on BSSH' % args.run_name)
+                t=1
+                break
+        except:
+            continue
+        try:
+            if tmp.at[args.run_name,'Status']=='Failed' or tmp.at[args.run_name,'Status']=='Stopped' or tmp.at[args.run_name,'Status']=='Needs Attention' or tmp.at[args.run_name,'Status']=='Timed Out':
+                user=i
+                idd=tmp.at[args.run_name,'Id']
+                slack_message('%s has "Failed" or was unable to complete on BSSH; Script is aborting, check BSSH for more information' % args.run_name)
+                print('%s has "Failed" or was unable to complete on BSSH; Script is aborting, check BSSH for more information' % args.run_name)
+                sys.exit()
+        except:
+            continue
+    # Wait 20 min then look again.
+    if t==1:
+        break
+    time.sleep(1200)
+
+# Make dir that reads and anaylsis files will be saved to
+try:
+    os.makedirs("/Volumes/IDGenomics_NAS/WGS_Serotyping/%s/Sequencing_reads"% args.run_name)
+except:
+    print("Directory Already Exits, Make sure Reads are not already downloaded and delete Directory")
+    sys.exit()
```

Comment:
kk. So this script won't run if the directory already exists. It's what the prior one did, too.